File: Data Strctures/Merge two sorted linked lists.py
def mergeLists(head1, head2):
  if head1 is None and head2 is None:
    return None
  
  if head1 is None:
    return head2

  if head2 is None:
    return head1
  
  if head1.data < head2.data:
    smallerNode = head1
    smallerNode.next = mergeLists(head1.next, head2)
  else:
    smallerNode = head2
    smallerNode.next = mergeLists(head1, head2.next)
  
  return smallerNode

File: Data Strctures/Merge two sorted linked lists.cc
SinglyLinkedListNode* mergeLists(SinglyLinkedListNode* head1, SinglyLinkedListNode* head2) {
    // If one of the linked lists is empty, return the other one 
    if (!head1)
        return head2;
    if (!head2)
        return head1;
    // Start with the linked list whose data is less than the other
    if (head1->data < head2->data) {
        head1->next = mergeLists(head1->next, head2);
        return head1;
    }
    head2->next = mergeLists(head1, head2->next);
    return head2;
}

//try with c++
